3rd_party/3rd_party.cmake

```diff
@@ -38,11 +38,11 @@ set(ARCH ${HOST_SYSTEM_PROCESSOR})
 set(CMAKE_SYSTEM_NAME ${HOST_SYSTEM_NAME})
 
 message(STATUS "CMAKE_SYSTEM_NAME = ${CMAKE_SYSTEM_NAME}")
-if (CMAKE_SYSTEM_NAME STREQUAL "darwin")
+if ("${CMAKE_SYSTEM_NAME}" STREQUAL "darwin")

```

Comment:
In many other files the `if (variable STREQUAL "literal")` syntax is used.  Why change to `if ("${variable}" STREQUAL "literal")` here?  I think it would be best to find out which is considered best practice and use the same style everywhere.